### ISSUE TITLE
[Bug Report] v-btn background color is black when icon = true

### ISSUE THREAD

**Issue by goeh** [vuetifyjs/vuetify#12945]
### Environment
**Vuetify Version:** 2.4.2
**Last working version:** 2.3.23
**Vue Version:** 2.6.12
**Browsers:** Chrome 87.0.4280.141
**OS:** Mac OS 11.1.0

### Steps to reproduce
      <v-toolbar dark color="primary">
            <v-toolbar-title>
              <v-icon left>mdi-account-multiple-outline</v-icon>
              Button background issue
            </v-toolbar-title>
            <v-spacer></v-spacer>
            <v-toolbar-items>
              <v-btn icon>
                <v-icon>mdi-plus</v-icon>
              </v-btn>
            </v-toolbar-items>
          </v-toolbar>

### Expected Behavior
In < 2.4.0 button background is transparent/inherited when type is "icon".

### Actual Behavior
In 2.4.2 button shows black rectangle behind icon.

### Reproduction Link
<a href="https://codepen.io/goeh/pen/ZEEeZbz" target="_blank">https://codepen.io/goeh/pen/ZEEeZbz</a>

### Other comments
Related #12903

**Comment by MajesticPotatoe** [comment]
Your codepen is invalid. You are using a mismatch of css + vuetify version
<https://codepen.io/majesticpotatoe-the-bashful/pen/YzGOgaY>

other than that it looks fine

**Comment by goeh** [comment]
@MajesticPotatoe Sorry I modified an old pen and forgot to change the CSS.
But I have this issue in an application I upgraded from 2.3.23 to 2.4.2. I need to check why the CSS is not working. This has never happened before when I upgrade.

**Comment by goeh** [comment]
If I fix the CodePen to use version 2.4.2 both for JS and CSS the problem goes away. But I can't get my application to work. It's driving me nuts. I will close this issue as soon as I find the problem in my application.

**Comment by goeh** [comment]
This is the style that is applied to my `<v-btn icon>` and renders the button background dark.
```
.theme--dark.v-btn:not(.v-btn--flat):not(.v-btn--text):not(.v-btn--outlined) {
    background-color: #272727;
}
```
Is that CSS from Vuetify 2.3.23 or from 2.4.2 ?

**Comment by goeh** [comment]
Not a bug. It was a CSS cache/mixup issue in my application. My application is modularized with four different packages (feature1, feature2, feature3, feature4). Each package is a Vuetify project with views and components for that specific feature.
I upgraded the main application from 2.3.23 to 4.2.4 but the feature packages was still compiled with Vuetify 2.3.23. Each feature project had the following import statements in plugin.js (that is read by the main application when "installing" each feature plugin).
```
import Vue from 'vue'
import Vuetify from 'vuetify/lib'
import 'vuetify/dist/vuetify.min.css'
```
So it seems that when the main application initializes feature plugins it imports CSS from vuetify 2.3.23.
I commented out the line `import 'vuetify/dist/vuetify.min.css' in each feature project and made new bundles. When the main application uses updated feature plugins the CSS seems to be fine and the button background color is correct.

The main application has been modularized like this for at least a year but I have never stumbled upon an issue like this. Maybe it has been small CSS issues that I haven't noticed.
